<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Exercises | Normalization (BCB420 Lecture)</title>
  <meta name="description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Exercises | Normalization (BCB420 Lecture)" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Exercises | Normalization (BCB420 Lecture)" />
  
  <meta name="twitter:description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  

<meta name="author" content="Ruth Isserlin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="worked-normalization-examples.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BCB420 Lecture 5</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>0.1</b> Learning goals</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i><b>0.2</b> Example datasets</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-normalization.html"><a href="what-is-normalization.html"><i class="fa fa-check"></i><b>1</b> What is normalization?</a></li>
<li class="chapter" data-level="2" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html"><i class="fa fa-check"></i><b>2</b> Technical vs biological variation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html#technical-variation"><i class="fa fa-check"></i><b>2.1</b> Technical variation</a></li>
<li class="chapter" data-level="2.2" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html#biological-variation"><i class="fa fa-check"></i><b>2.2</b> Biological variation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html"><i class="fa fa-check"></i><b>3</b> Visualizing distributions</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#load-some-data"><i class="fa fa-check"></i><b>3.1</b> load Some data</a></li>
<li class="chapter" data-level="3.2" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#boxplots"><i class="fa fa-check"></i><b>3.2</b> Boxplots</a></li>
<li class="chapter" data-level="3.3" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#density-plots"><i class="fa fa-check"></i><b>3.3</b> Density plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-methods.html"><a href="normalization-methods.html"><i class="fa fa-check"></i><b>4</b> Normalization methods</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-methods.html"><a href="normalization-methods.html#library-size-scaling"><i class="fa fa-check"></i><b>4.1</b> Library-size scaling</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="normalization-methods.html"><a href="normalization-methods.html#counts-per-million-cpm"><i class="fa fa-check"></i><b>4.1.1</b> Counts-per-million (CPM)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="normalization-methods.html"><a href="normalization-methods.html#distribution-based-normalization"><i class="fa fa-check"></i><b>4.2</b> Distribution-based normalization</a></li>
<li class="chapter" data-level="4.3" data-path="normalization-methods.html"><a href="normalization-methods.html#rna-seq-composition-aware-methods"><i class="fa fa-check"></i><b>4.3</b> RNA-seq composition-aware methods</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="normalization-methods.html"><a href="normalization-methods.html#edger-tmm"><i class="fa fa-check"></i><b>4.3.1</b> edgeR: TMM</a></li>
<li class="chapter" data-level="4.3.2" data-path="normalization-methods.html"><a href="normalization-methods.html#deseq2-size-factors"><i class="fa fa-check"></i><b>4.3.2</b> DESeq2: size factors</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="normalization-methods.html"><a href="normalization-methods.html#single-cell-note"><i class="fa fa-check"></i><b>4.4</b> Single-cell note</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html"><i class="fa fa-check"></i><b>5</b> Worked normalization examples</a>
<ul>
<li class="chapter" data-level="5.1" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#setup-download-supplementary-files"><i class="fa fa-check"></i><b>5.1</b> Setup: download supplementary files</a></li>
<li class="chapter" data-level="5.2" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#load-some-data-1"><i class="fa fa-check"></i><b>5.2</b> load Some data</a></li>
<li class="chapter" data-level="5.3" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#example-1-gse119732"><i class="fa fa-check"></i><b>5.3</b> Example 1: GSE119732</a></li>
<li class="chapter" data-level="5.4" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#convert-the-raw-counts-to-counts-per-million-cpm"><i class="fa fa-check"></i><b>5.4</b> Convert the raw counts to Counts per million (CPM)</a></li>
<li class="chapter" data-level="5.5" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#filter-out-lowly-expressed-genes"><i class="fa fa-check"></i><b>5.5</b> filter out lowly expressed genes</a></li>
<li class="chapter" data-level="5.6" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#incorporate-a-design-matrix---description-of-the-samples."><i class="fa fa-check"></i><b>5.6</b> Incorporate a design matrix - description of the samples.</a></li>
<li class="chapter" data-level="5.7" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#normalize-dataset-using-tmm"><i class="fa fa-check"></i><b>5.7</b> Normalize Dataset using TMM</a></li>
<li class="chapter" data-level="5.8" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#look-at-the-distribution-of-our-samples-in-2d-space"><i class="fa fa-check"></i><b>5.8</b> Look at the distribution of our samples in 2D space</a></li>
<li class="chapter" data-level="5.9" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#normalize-the-dataset-using-rle"><i class="fa fa-check"></i><b>5.9</b> Normalize the Dataset using RLE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>6</b> Exercises</a>
<ul>
<li class="chapter" data-level="6.1" data-path="exercises.html"><a href="exercises.html#exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732"><i class="fa fa-check"></i><b>6.1</b> Exercise 1 — https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="exercises.html"><a href="exercises.html#helper-functions"><i class="fa fa-check"></i><b>6.1.1</b> Helper functions</a></li>
<li class="chapter" data-level="6.1.2" data-path="exercises.html"><a href="exercises.html#question-1"><i class="fa fa-check"></i><b>6.1.2</b> Question 1</a></li>
<li class="chapter" data-level="6.1.3" data-path="exercises.html"><a href="exercises.html#question-2"><i class="fa fa-check"></i><b>6.1.3</b> Question 2</a></li>
<li class="chapter" data-level="6.1.4" data-path="exercises.html"><a href="exercises.html#question-2-1"><i class="fa fa-check"></i><b>6.1.4</b> Question 2</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="exercises.html"><a href="exercises.html#exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380"><i class="fa fa-check"></i><b>6.2</b> Exercise 2 — https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="exercises.html"><a href="exercises.html#question-1-1"><i class="fa fa-check"></i><b>6.2.1</b> Question 1</a></li>
<li class="chapter" data-level="6.2.2" data-path="exercises.html"><a href="exercises.html#question-2-2"><i class="fa fa-check"></i><b>6.2.2</b> Question 2</a></li>
<li class="chapter" data-level="6.2.3" data-path="exercises.html"><a href="exercises.html#question-2-3"><i class="fa fa-check"></i><b>6.2.3</b> Question 2</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="exercises.html"><a href="exercises.html#exercise-3--"><i class="fa fa-check"></i><b>6.3</b> Exercise 3 -</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Normalization (BCB420 Lecture)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exercises" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Exercises<a href="exercises.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>These are intended to be done <strong>after</strong> completing the worked examples.</p>
<div id="exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Exercise 1 — <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732</a><a href="exercises.html#exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <strong>GSE119732</strong>,</p>
<ol style="list-style-type: decimal">
<li>Map the distribution of the samples as a box plot and a density plot using only the raw data.</li>
<li>Add colours to the plot separating the different variables in the model</li>
<li>Filter out lowly expressed genes and re plot the distributions.</li>
</ol>
<div id="helper-functions" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Helper functions<a href="exercises.html#helper-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="exercises.html#cb55-1" tabindex="-1"></a><span class="co"># Load library</span></span>
<span id="cb55-2"><a href="exercises.html#cb55-2" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb55-3"><a href="exercises.html#cb55-3" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb55-4"><a href="exercises.html#cb55-4" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="exercises.html#cb59-1" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb59-2"><a href="exercises.html#cb59-2" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span></code></pre></div>
<pre><code>## Loading required package: limma</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="exercises.html#cb61-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb61-2"><a href="exercises.html#cb61-2" tabindex="-1"></a></span>
<span id="cb61-3"><a href="exercises.html#cb61-3" tabindex="-1"></a><span class="co"># Safe read files</span></span>
<span id="cb61-4"><a href="exercises.html#cb61-4" tabindex="-1"></a>safe_read <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb61-5"><a href="exercises.html#cb61-5" tabindex="-1"></a>  <span class="co"># First attempt: read as TSV</span></span>
<span id="cb61-6"><a href="exercises.html#cb61-6" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(</span>
<span id="cb61-7"><a href="exercises.html#cb61-7" tabindex="-1"></a>    readr<span class="sc">::</span><span class="fu">read_tsv</span>(file, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>),</span>
<span id="cb61-8"><a href="exercises.html#cb61-8" tabindex="-1"></a>    <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span>   <span class="co"># catch fatal errors</span></span>
<span id="cb61-9"><a href="exercises.html#cb61-9" tabindex="-1"></a>  )</span>
<span id="cb61-10"><a href="exercises.html#cb61-10" tabindex="-1"></a>  </span>
<span id="cb61-11"><a href="exercises.html#cb61-11" tabindex="-1"></a>  <span class="co"># If read_tsv failed entirely:</span></span>
<span id="cb61-12"><a href="exercises.html#cb61-12" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(df)) {</span>
<span id="cb61-13"><a href="exercises.html#cb61-13" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;TSV read failed — reading as space-delimited file instead.&quot;</span>)</span>
<span id="cb61-14"><a href="exercises.html#cb61-14" tabindex="-1"></a>    <span class="fu">return</span>(readr<span class="sc">::</span><span class="fu">read_table</span>(file, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>))</span>
<span id="cb61-15"><a href="exercises.html#cb61-15" tabindex="-1"></a>  }</span>
<span id="cb61-16"><a href="exercises.html#cb61-16" tabindex="-1"></a>  </span>
<span id="cb61-17"><a href="exercises.html#cb61-17" tabindex="-1"></a>  <span class="co"># If read_tsv returned but with parsing issues:</span></span>
<span id="cb61-18"><a href="exercises.html#cb61-18" tabindex="-1"></a>  probs <span class="ot">&lt;-</span> <span class="fu">problems</span>(df)</span>
<span id="cb61-19"><a href="exercises.html#cb61-19" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(probs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb61-20"><a href="exercises.html#cb61-20" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;Parsing issues detected in TSV — reading as space-delimited file instead.&quot;</span>)</span>
<span id="cb61-21"><a href="exercises.html#cb61-21" tabindex="-1"></a>    <span class="fu">return</span>(readr<span class="sc">::</span><span class="fu">read_table</span>(file, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>))</span>
<span id="cb61-22"><a href="exercises.html#cb61-22" tabindex="-1"></a>  }</span>
<span id="cb61-23"><a href="exercises.html#cb61-23" tabindex="-1"></a>  </span>
<span id="cb61-24"><a href="exercises.html#cb61-24" tabindex="-1"></a>  <span class="co"># If everything was fine:</span></span>
<span id="cb61-25"><a href="exercises.html#cb61-25" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb61-26"><a href="exercises.html#cb61-26" tabindex="-1"></a>}</span>
<span id="cb61-27"><a href="exercises.html#cb61-27" tabindex="-1"></a></span>
<span id="cb61-28"><a href="exercises.html#cb61-28" tabindex="-1"></a></span>
<span id="cb61-29"><a href="exercises.html#cb61-29" tabindex="-1"></a><span class="co"># Box Plot to support coloring by variable</span></span>
<span id="cb61-30"><a href="exercises.html#cb61-30" tabindex="-1"></a>plot_box_by_var <span class="ot">&lt;-</span> <span class="cf">function</span>(mat, meta, fill_var, <span class="at">main =</span> <span class="st">&quot;&quot;</span>) {</span>
<span id="cb61-31"><a href="exercises.html#cb61-31" tabindex="-1"></a>  df_long <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mat) <span class="sc">%&gt;%</span></span>
<span id="cb61-32"><a href="exercises.html#cb61-32" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(mat)) <span class="sc">%&gt;%</span></span>
<span id="cb61-33"><a href="exercises.html#cb61-33" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-34"><a href="exercises.html#cb61-34" tabindex="-1"></a>    <span class="fu">left_join</span>(meta, <span class="at">by =</span> <span class="st">&quot;samples&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-35"><a href="exercises.html#cb61-35" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">log2</span>(value <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb61-36"><a href="exercises.html#cb61-36" tabindex="-1"></a>  </span>
<span id="cb61-37"><a href="exercises.html#cb61-37" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> samples, <span class="at">y =</span> value, <span class="at">fill =</span> .data[[fill_var]])) <span class="sc">+</span></span>
<span id="cb61-38"><a href="exercises.html#cb61-38" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">outlier.size =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb61-39"><a href="exercises.html#cb61-39" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb61-40"><a href="exercises.html#cb61-40" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb61-41"><a href="exercises.html#cb61-41" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> main, <span class="at">y =</span> <span class="st">&quot;log2(counts+1)&quot;</span>)</span>
<span id="cb61-42"><a href="exercises.html#cb61-42" tabindex="-1"></a>}</span>
<span id="cb61-43"><a href="exercises.html#cb61-43" tabindex="-1"></a></span>
<span id="cb61-44"><a href="exercises.html#cb61-44" tabindex="-1"></a></span>
<span id="cb61-45"><a href="exercises.html#cb61-45" tabindex="-1"></a><span class="co"># Density Plot to support coloring by variable</span></span>
<span id="cb61-46"><a href="exercises.html#cb61-46" tabindex="-1"></a>plot_density_by_var <span class="ot">&lt;-</span> <span class="cf">function</span>(mat, meta, color_var, <span class="at">main =</span> <span class="st">&quot;&quot;</span>) {</span>
<span id="cb61-47"><a href="exercises.html#cb61-47" tabindex="-1"></a>  df_long <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mat) <span class="sc">%&gt;%</span></span>
<span id="cb61-48"><a href="exercises.html#cb61-48" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(mat)) <span class="sc">%&gt;%</span></span>
<span id="cb61-49"><a href="exercises.html#cb61-49" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-50"><a href="exercises.html#cb61-50" tabindex="-1"></a>    <span class="fu">left_join</span>(meta, <span class="at">by =</span> <span class="st">&quot;samples&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-51"><a href="exercises.html#cb61-51" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">log2</span>(value <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb61-52"><a href="exercises.html#cb61-52" tabindex="-1"></a>  </span>
<span id="cb61-53"><a href="exercises.html#cb61-53" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">color =</span> .data[[color_var]], <span class="at">group =</span> samples)) <span class="sc">+</span></span>
<span id="cb61-54"><a href="exercises.html#cb61-54" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb61-55"><a href="exercises.html#cb61-55" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb61-56"><a href="exercises.html#cb61-56" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> main, <span class="at">x =</span> <span class="st">&quot;log2(counts+1)&quot;</span>)</span>
<span id="cb61-57"><a href="exercises.html#cb61-57" tabindex="-1"></a>}</span>
<span id="cb61-58"><a href="exercises.html#cb61-58" tabindex="-1"></a></span>
<span id="cb61-59"><a href="exercises.html#cb61-59" tabindex="-1"></a></span>
<span id="cb61-60"><a href="exercises.html#cb61-60" tabindex="-1"></a><span class="co"># Load existing helper function</span></span>
<span id="cb61-61"><a href="exercises.html#cb61-61" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./supp_functions.R&quot;</span>)</span></code></pre></div>
</div>
<div id="question-1" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Question 1<a href="exercises.html#question-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="raw-box-plot" class="section level4 hasAnchor" number="6.1.2.1">
<h4><span class="header-section-number">6.1.2.1</span> Raw Box Plot<a href="exercises.html#raw-box-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="exercises.html#cb62-1" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="st">&quot;GSE119732&quot;</span></span>
<span id="cb62-2"><a href="exercises.html#cb62-2" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;GSE119732&quot;</span>)</span>
<span id="cb62-3"><a href="exercises.html#cb62-3" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.txt.gz$|</span><span class="sc">\\</span><span class="st">.tsv.gz$|</span><span class="sc">\\</span><span class="st">.csv.gz$&quot;</span>, </span>
<span id="cb62-4"><a href="exercises.html#cb62-4" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb62-5"><a href="exercises.html#cb62-5" tabindex="-1"></a></span>
<span id="cb62-6"><a href="exercises.html#cb62-6" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">safe_read</span>(files[<span class="dv">1</span>])</span>
<span id="cb62-7"><a href="exercises.html#cb62-7" tabindex="-1"></a></span>
<span id="cb62-8"><a href="exercises.html#cb62-8" tabindex="-1"></a><span class="fu">plot_box</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)], <span class="at">main =</span> <span class="st">&quot;GSE119732 Raw Boxplot&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-2-1.png" alt="" width="672" /></p>
</div>
<div id="raw-density-plot" class="section level4 hasAnchor" number="6.1.2.2">
<h4><span class="header-section-number">6.1.2.2</span> Raw Density Plot<a href="exercises.html#raw-density-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="exercises.html#cb63-1" tabindex="-1"></a><span class="fu">plot_density</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)], <span class="at">main =</span> <span class="st">&quot;GSE119732 Raw Density&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-3-1.png" alt="" width="672" /></p>
</div>
</div>
<div id="question-2" class="section level3 hasAnchor" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Question 2<a href="exercises.html#question-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="boxplot-by-patient" class="section level4 hasAnchor" number="6.1.3.1">
<h4><span class="header-section-number">6.1.3.1</span> Boxplot by patient<a href="exercises.html#boxplot-by-patient" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="exercises.html#cb64-1" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)]</span>
<span id="cb64-2"><a href="exercises.html#cb64-2" tabindex="-1"></a>patient  <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples, <span class="dv">1</span>, <span class="dv">1</span>)       </span>
<span id="cb64-3"><a href="exercises.html#cb64-3" tabindex="-1"></a>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(samples, <span class="dv">2</span>, <span class="fu">nchar</span>(samples))   </span>
<span id="cb64-4"><a href="exercises.html#cb64-4" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(samples, patient, celltype)</span>
<span id="cb64-5"><a href="exercises.html#cb64-5" tabindex="-1"></a></span>
<span id="cb64-6"><a href="exercises.html#cb64-6" tabindex="-1"></a>raw_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span>
<span id="cb64-7"><a href="exercises.html#cb64-7" tabindex="-1"></a><span class="fu">rownames</span>(raw_mat) <span class="ot">&lt;-</span> x[[<span class="dv">1</span>]]</span>
<span id="cb64-8"><a href="exercises.html#cb64-8" tabindex="-1"></a></span>
<span id="cb64-9"><a href="exercises.html#cb64-9" tabindex="-1"></a><span class="co"># Generate the plots</span></span>
<span id="cb64-10"><a href="exercises.html#cb64-10" tabindex="-1"></a><span class="fu">plot_box_by_var</span>(raw_mat, sample_data, <span class="st">&quot;patient&quot;</span>, <span class="at">main =</span> <span class="st">&quot;GSE119732 Raw: Boxplot by Patient&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-4-1.png" alt="" width="672" /></p>
</div>
<div id="density-by-patient" class="section level4 hasAnchor" number="6.1.3.2">
<h4><span class="header-section-number">6.1.3.2</span> Density by patient<a href="exercises.html#density-by-patient" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="exercises.html#cb65-1" tabindex="-1"></a><span class="fu">plot_density_by_var</span>(raw_mat, sample_data, <span class="st">&quot;patient&quot;</span>, <span class="at">main =</span> <span class="st">&quot;GSE119732 Raw: Density by Patient&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-5-1.png" alt="" width="672" /></p>
</div>
<div id="boxplot-by-cell-type" class="section level4 hasAnchor" number="6.1.3.3">
<h4><span class="header-section-number">6.1.3.3</span> Boxplot by cell type<a href="exercises.html#boxplot-by-cell-type" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="exercises.html#cb66-1" tabindex="-1"></a><span class="fu">plot_box_by_var</span>(raw_mat, sample_data, <span class="st">&quot;celltype&quot;</span>, <span class="st">&quot;GSE119732 Raw: Boxplot by Cell Type&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-6-1.png" alt="" width="672" /></p>
</div>
<div id="density-by-cell-type" class="section level4 hasAnchor" number="6.1.3.4">
<h4><span class="header-section-number">6.1.3.4</span> Density by cell type<a href="exercises.html#density-by-cell-type" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="exercises.html#cb67-1" tabindex="-1"></a><span class="fu">plot_density_by_var</span>(raw_mat, sample_data, <span class="st">&quot;celltype&quot;</span>, <span class="st">&quot;GSE119732 Raw: Density by Cell Type&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-7-1.png" alt="" width="672" /></p>
</div>
</div>
<div id="question-2-1" class="section level3 hasAnchor" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> Question 2<a href="exercises.html#question-2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="boxplot-with-expression-3-removed" class="section level4 hasAnchor" number="6.1.4.1">
<h4><span class="header-section-number">6.1.4.1</span> Boxplot with expression &lt;3 removed<a href="exercises.html#boxplot-with-expression-3-removed" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="exercises.html#cb68-1" tabindex="-1"></a><span class="co"># Convert the raw counts to Counts per million</span></span>
<span id="cb68-2"><a href="exercises.html#cb68-2" tabindex="-1"></a>x_cpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(<span class="at">y =</span> x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span>
<span id="cb68-3"><a href="exercises.html#cb68-3" tabindex="-1"></a></span>
<span id="cb68-4"><a href="exercises.html#cb68-4" tabindex="-1"></a><span class="co"># remove low expression ones</span></span>
<span id="cb68-5"><a href="exercises.html#cb68-5" tabindex="-1"></a>to_remove <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">filterByExpr</span>(x_cpm, <span class="at">min.count =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## No group or design set. Assuming all samples belong to one group.</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="exercises.html#cb70-1" tabindex="-1"></a>x_cpm_filtered <span class="ot">&lt;-</span> x_cpm[to_remove,]</span>
<span id="cb70-2"><a href="exercises.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="exercises.html#cb70-3" tabindex="-1"></a><span class="fu">plot_box</span>(x_cpm_filtered, <span class="at">main =</span> <span class="st">&quot;GSE119732 Boxplot: Filtered out lowly expressed&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-8-1.png" alt="" width="672" /></p>
</div>
<div id="density-with-expression-3-removed" class="section level4 hasAnchor" number="6.1.4.2">
<h4><span class="header-section-number">6.1.4.2</span> Density with expression &lt;3 removed<a href="exercises.html#density-with-expression-3-removed" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="exercises.html#cb71-1" tabindex="-1"></a><span class="fu">plot_density</span>(x_cpm_filtered, <span class="at">main =</span> <span class="st">&quot;GSE119732 Density: Filtered out lowly expressed&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-9-1.png" alt="" width="672" /></p>
</div>
</div>
</div>
<div id="exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Exercise 2 — <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380</a><a href="exercises.html#exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <strong>GSE122380</strong>, confirm whether the ID column contains Ensembl IDs with version suffixes.</p>
<ol style="list-style-type: decimal">
<li>Map the distribution of the samples as a box plot and a density plot using only the raw data.</li>
<li>Add colours to the plot separating the different variables in the model</li>
<li>Filter out lowly expressed genes and re plot the distributions.</li>
</ol>
<div id="question-1-1" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Question 1<a href="exercises.html#question-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="check-for-version-suffixes" class="section level4 hasAnchor" number="6.2.1.1">
<h4><span class="header-section-number">6.2.1.1</span> Check for version suffixes<a href="exercises.html#check-for-version-suffixes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="exercises.html#cb72-1" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="st">&quot;GSE122380&quot;</span></span>
<span id="cb72-2"><a href="exercises.html#cb72-2" tabindex="-1"></a><span class="fu">fetch_geo_supp</span>(<span class="at">gse =</span> gse)</span></code></pre></div>
<pre><code>## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)</code></pre>
<pre><code>## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)</code></pre>
<pre><code>## Using locally cached version of supplementary file(s) GSE122380 found here:
## data/GSE122380/GSE122380_Supplementary_Data_Table_S1.xlsx</code></pre>
<pre><code>## Using locally cached version of supplementary file(s) GSE122380 found here:
## data/GSE122380/GSE122380_raw_counts.txt.gz</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="exercises.html#cb77-1" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;GSE122380&quot;</span>)</span>
<span id="cb77-2"><a href="exercises.html#cb77-2" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.txt.gz$|</span><span class="sc">\\</span><span class="st">.tsv.gz$|</span><span class="sc">\\</span><span class="st">.csv.gz$&quot;</span>, </span>
<span id="cb77-3"><a href="exercises.html#cb77-3" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb77-4"><a href="exercises.html#cb77-4" tabindex="-1"></a></span>
<span id="cb77-5"><a href="exercises.html#cb77-5" tabindex="-1"></a>x<span class="ot">&lt;-</span> <span class="fu">safe_read</span>(files[<span class="dv">1</span>])</span></code></pre></div>
<pre><code>## Warning: One or more parsing issues, call `problems()` on your data frame for details,
## e.g.:
##   dat &lt;- vroom(...)
##   problems(dat)</code></pre>
<pre><code>## Parsing issues detected in TSV — reading as space-delimited file instead.</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="exercises.html#cb80-1" tabindex="-1"></a><span class="co"># Count how many total contain a &quot;.&quot;</span></span>
<span id="cb80-2"><a href="exercises.html#cb80-2" tabindex="-1"></a>dot_count <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, x<span class="sc">$</span>gene_id))</span></code></pre></div>
<pre><code>## Warning: Unknown or uninitialised column: `gene_id`.</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="exercises.html#cb82-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Total number of IDs with version suffixes:&quot;</span>, dot_count))</span></code></pre></div>
<pre><code>## [1] &quot;Total number of IDs with version suffixes: 0&quot;</code></pre>
<p>ID column <strong>does not</strong> contains Ensembl IDs with version suffixes.</p>
</div>
<div id="raw-density-plot-1" class="section level4 hasAnchor" number="6.2.1.2">
<h4><span class="header-section-number">6.2.1.2</span> Raw Density Plot<a href="exercises.html#raw-density-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="exercises.html#cb84-1" tabindex="-1"></a><span class="fu">plot_box</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)], <span class="at">main =</span> <span class="st">&quot;GSE122380 Raw Boxplot&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-11-1.png" alt="" width="672" /></p>
</div>
<div id="raw-density-plot-2" class="section level4 hasAnchor" number="6.2.1.3">
<h4><span class="header-section-number">6.2.1.3</span> Raw Density Plot<a href="exercises.html#raw-density-plot-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="exercises.html#cb85-1" tabindex="-1"></a><span class="fu">plot_density</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)], <span class="at">main =</span> <span class="st">&quot;GSE122380 Raw Density&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-12-1.png" alt="" width="672" /></p>
</div>
</div>
<div id="question-2-2" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Question 2<a href="exercises.html#question-2-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="boxplot-by-patient-1" class="section level4 hasAnchor" number="6.2.2.1">
<h4><span class="header-section-number">6.2.2.1</span> Boxplot by patient<a href="exercises.html#boxplot-by-patient-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="exercises.html#cb86-1" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)]</span>
<span id="cb86-2"><a href="exercises.html#cb86-2" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples =</span> samples) <span class="sc">%&gt;%</span></span>
<span id="cb86-3"><a href="exercises.html#cb86-3" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb86-4"><a href="exercises.html#cb86-4" tabindex="-1"></a>    <span class="at">patient =</span> <span class="fu">sub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, samples),</span>
<span id="cb86-5"><a href="exercises.html#cb86-5" tabindex="-1"></a>    <span class="at">timepoint =</span> <span class="fu">sub</span>(<span class="st">&quot;.*_&quot;</span>, <span class="st">&quot;&quot;</span>, samples)</span>
<span id="cb86-6"><a href="exercises.html#cb86-6" tabindex="-1"></a>  )</span>
<span id="cb86-7"><a href="exercises.html#cb86-7" tabindex="-1"></a>sample_data<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> <span class="fu">factor</span>(sample_data<span class="sc">$</span>timepoint, </span>
<span id="cb86-8"><a href="exercises.html#cb86-8" tabindex="-1"></a>                                <span class="at">levels =</span> <span class="fu">as.character</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">15</span>))</span>
<span id="cb86-9"><a href="exercises.html#cb86-9" tabindex="-1"></a></span>
<span id="cb86-10"><a href="exercises.html#cb86-10" tabindex="-1"></a>raw_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span>
<span id="cb86-11"><a href="exercises.html#cb86-11" tabindex="-1"></a><span class="fu">rownames</span>(raw_mat) <span class="ot">&lt;-</span> x[[<span class="dv">1</span>]]</span>
<span id="cb86-12"><a href="exercises.html#cb86-12" tabindex="-1"></a></span>
<span id="cb86-13"><a href="exercises.html#cb86-13" tabindex="-1"></a><span class="co"># Generate the plots</span></span>
<span id="cb86-14"><a href="exercises.html#cb86-14" tabindex="-1"></a><span class="fu">plot_box_by_var</span>(raw_mat, sample_data, <span class="st">&quot;patient&quot;</span>, <span class="at">main =</span> <span class="st">&quot;GSE122380 Raw: Boxplot by Patient&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-13-1.png" alt="" width="672" /></p>
</div>
<div id="density-by-patient-1" class="section level4 hasAnchor" number="6.2.2.2">
<h4><span class="header-section-number">6.2.2.2</span> Density by patient<a href="exercises.html#density-by-patient-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="exercises.html#cb87-1" tabindex="-1"></a><span class="fu">plot_density_by_var</span>(raw_mat, sample_data, <span class="st">&quot;patient&quot;</span>, <span class="at">main =</span> <span class="st">&quot;GSE122380 Raw: Density by Patient&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-14-1.png" alt="" width="672" /></p>
</div>
<div id="boxplot-by-timepoint" class="section level4 hasAnchor" number="6.2.2.3">
<h4><span class="header-section-number">6.2.2.3</span> Boxplot by Timepoint<a href="exercises.html#boxplot-by-timepoint" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="exercises.html#cb88-1" tabindex="-1"></a><span class="fu">plot_box_by_var</span>(raw_mat, sample_data, <span class="st">&quot;timepoint&quot;</span>, <span class="st">&quot;GSE122380 Raw: Boxplot by Timepoint&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-15-1.png" alt="" width="672" /></p>
</div>
<div id="density-by-timepoint" class="section level4 hasAnchor" number="6.2.2.4">
<h4><span class="header-section-number">6.2.2.4</span> Density by Timepoint<a href="exercises.html#density-by-timepoint" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="exercises.html#cb89-1" tabindex="-1"></a><span class="fu">plot_density_by_var</span>(raw_mat, sample_data, <span class="st">&quot;timepoint&quot;</span>, <span class="st">&quot;GSE122380 Raw: Density by Timepoint&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-16-1.png" alt="" width="672" /></p>
</div>
</div>
<div id="question-2-3" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Question 2<a href="exercises.html#question-2-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="boxplot-with-expression-3-removed-1" class="section level4 hasAnchor" number="6.2.3.1">
<h4><span class="header-section-number">6.2.3.1</span> Boxplot with expression &lt;3 removed<a href="exercises.html#boxplot-with-expression-3-removed-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="exercises.html#cb90-1" tabindex="-1"></a><span class="co"># Convert the raw counts to Counts per million</span></span>
<span id="cb90-2"><a href="exercises.html#cb90-2" tabindex="-1"></a>x_cpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(<span class="at">y =</span> x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span>
<span id="cb90-3"><a href="exercises.html#cb90-3" tabindex="-1"></a></span>
<span id="cb90-4"><a href="exercises.html#cb90-4" tabindex="-1"></a><span class="co"># remove low expression ones</span></span>
<span id="cb90-5"><a href="exercises.html#cb90-5" tabindex="-1"></a>to_remove <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">filterByExpr</span>(x_cpm, <span class="at">min.count =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## No group or design set. Assuming all samples belong to one group.</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="exercises.html#cb92-1" tabindex="-1"></a>x_cpm_filtered <span class="ot">&lt;-</span> x_cpm[to_remove,]</span>
<span id="cb92-2"><a href="exercises.html#cb92-2" tabindex="-1"></a></span>
<span id="cb92-3"><a href="exercises.html#cb92-3" tabindex="-1"></a><span class="fu">plot_box</span>(x_cpm_filtered, <span class="at">main =</span> <span class="st">&quot;GSE122380 Boxplot: Filtered out lowly expressed&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-17-1.png" alt="" width="672" /></p>
</div>
<div id="density-with-expression-3-removed-1" class="section level4 hasAnchor" number="6.2.3.2">
<h4><span class="header-section-number">6.2.3.2</span> Density with expression &lt;3 removed<a href="exercises.html#density-with-expression-3-removed-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="exercises.html#cb93-1" tabindex="-1"></a><span class="fu">plot_density</span>(x_cpm_filtered, <span class="at">main =</span> <span class="st">&quot;GSE122380 Density: Filtered out lowly expressed&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-18-1.png" alt="" width="672" /></p>
</div>
</div>
</div>
<div id="exercise-3--" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Exercise 3 -<a href="exercises.html#exercise-3--" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Can you use the worked example to process the above two GEO records? How?</p>
<p>Yes, but need modifications such as creating a design to distinguish variable
and might need stripping version numbers depends on the data type.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="worked-normalization-examples.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["bcb420_normalization.pdf", "bcb420_normalization.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
